src/libraries/Native/AnyOS/System.Globalization.Native/pal_collation.c

```diff
@@ -403,8 +407,12 @@ static const UCollator* GetCollatorFromSortHandle(SortHandle* pSortHandle, int32
         pCollator = CloneCollatorWithOptions(pSortHandle->collatorsPerOption[0], options, pErr);
         UCollator* pNull = NULL;
 
+#ifdef TARGET_UNIX
         // we are not using the standard atomic_compare_exchange_strong to workaround bugs in clang 5.0 (https://bugs.llvm.org/show_bug.cgi?id=37457)
         if (!__atomic_compare_exchange_n(&pSortHandle->collatorsPerOption[options], &pNull, pCollator, false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST))
+#else
+        if (!::std::atomic_compare_exchange_strong_explicit((volatile ::std::atomic<UCollator*> *const)&pSortHandle->collatorsPerOption[options], &pNull, pCollator, ::std::memory_order_seq_cst, ::std::memory_order_seq_cst))
```

Comment:
This looks like C++. C++ is not OK to use in the shims - it is not portable to mono. Use C equivalent instead.

Reply:
I tried and it couldn’t find the symbol. It seems like the atomic headers define it under the std namespace. What would the C equivalent be?

Reply:
http://www.cplusplus.com/reference/atomic/atomic_compare_exchange_strong/

Reply:
Ah ok, I see that you are using that already.

Reply:
Yeah, but the Windows <atomic> header defines the API under the std namespace. Do you have any ideas on what to here then?

Reply:
It is a mess. I guess it may be best to start `pal_atomic.h` where we implement what we need using specific compiler intrinsics.

Reply:
`mono_atomic_cas_ptr` from https://github.com/dotnet/runtime/blob/master/src/mono/mono/utils/atomic.h may be useful prior art.

Reply:
Why don't we use just InterlockedCompareExchangePointer  on Windows?

Reply:
Yeah. That is just what I'm doing with a helper function to avoid the if def here.